zerver/views/messages.py

```diff
@@ -1034,7 +1034,21 @@ def update_message_backend(request, user_profile,
                                        propagate_mode, content, rendered_content)
     # Include the number of messages changed in the logs
     request._log_data['extra'] = "[%s]" % (number_changed,)
-    if links_for_embed and getattr(settings, 'INLINE_URL_EMBED_PREVIEW', None):
+
+    current_message_realm_id = user_profile.realm
+    if not settings.INLINE_URL_EMBED_PREVIEW:
+        show_url_preview = False
+    else:
+        show_url_preview = True
+        if current_message_realm_id is not None:
+            try:
+                realm = Realm.objects.get(id=current_message_realm_id)
+            except:
+                realm = None
```

Comment:
This logic to compute the realm should just be `user_profile.realm` -- that is an object, not an ID.